### PR TITLE
move ReactiveQueue definition

### PR DESCRIPTION
**Pull request by chlebowa**
sister to [this PR](https://github.com/insightsengineering/teal.slice/pull/170)

Adds class definition for `Queue` and `ReactiveQueue` as well as unit tests.

### CODE REVIEW

**Review by gogonzo — APPROVED** [review]

File: R/Queue.R [hunk 1]
```diff
@@ -0,0 +1,179 @@
+# Queue ====
+#' @title  R6 Class - A First-In-First-Out Abstract Data Type
+#'
+#' @description `r lifecycle::badge("experimental")`
+#'
+#' @description
+#' Implements the queue abstract data type. The last element added to this queue is
+#' the last one to be returned from it.
+#' @export
```
Comment (gogonzo) [inline R/Queue.R:9]: I'm not sure whether we need to export this. I think it can stay internal as it's not used anywhere else

File: R/Queue.R [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+# ReactiveQueue ====
+#'
+#' @title  Reactive extension of \code{Queue} class
+#'
+#' @description
+#' Implements the queue abstract data type. The last element added to this queue is
+#' the last one to be returned from it.
+#' @keywords internal
+#'
+ReactiveQueue <- R6::R6Class( # nolint
```
Comment (gogonzo) [inline R/Queue.R:130]: ReactiveQueue can be removed

Comment (chlebowa) [reply R/Queue.R:130]: :+1:

File: R/Queue.R (hunk 1 above)
Comment (chlebowa) [reply R/Queue.R:9]: Ah, right, I forgot about that.

**Review by gogonzo — APPROVED** [review]

**Pull request merged**